test/realm_test.dart

```diff
@@ -552,5 +552,183 @@ Future<void> main([List<String>? args]) async {
       expect(() => realm.write(() => players[-1] = Person()), throws<RealmException>("Index out of range"));
       expect(() => realm.write(() => players[800] = Person()), throws<RealmException>());
     });
+
+    test('Realm RemoveMany from List', () {
+      var config = Configuration([Team.schema, Person.schema]);
+      var realm = Realm(config);
+
+      //Create two Teams
+      realm.write(() {
+        realm.add(Team()..name = "Ferrari");
+        realm.add(Team()..name = "Maserati");
+      });
+
+      //Ensure the teams exist in realm
+      var teams = realm.all<Team>();
+      expect(teams.length, 2);
+
+      //Convert RealmResults to List (asList should be implemented)
+      List<Team> list = [];
+      int itemsCount = teams.length;
+      for (var i = 0; i < itemsCount; i++) {
+        list.add(teams[i]);
+      }
+
+      //Remove objects in the list from realm
+      realm.write(() => realm.removeMany(list));
+
+      //Reload teams from realm and ensure they are removed
+      teams = realm.all<Team>();
```

Comment:
Yes, we discussed this with @nirinchev. We agreed that both could be checked the collection before reload and after reload. I have it in most of the test. If you think it is too much I will remove reloading from all the test.

Reply:
no its fine, I noticed the discussion. This comment was done before I knew it was discussed. Feel free do it as you see fit

Reply:
This test doesn't check whether the list is empty, because I don't clear the list. I only remove objects from realm database. We have to decide here, whether the user would expect the collection to be also cleared or not.